sbi/simulators/user_input_checks.py

```diff
@@ -172,10 +181,17 @@ def maybe_reinterpret_batch_dims(prior) -> Distribution:
     elif isinstance(prior, reinterpretable_priors) and prior.batch_shape.numel() > 1:
         warnings.warn(
             f"""The specified prior is a prior on *several scalar parameters*
-            (i.e., a batch), not a prior on a multi-dimensional parameter. 
+            (i.e., a batch), not a prior on a multi-dimensional parameter.
```

Comment:
I can't follow. for all elements of the set of reinterpretable_priors it is not possible to define them with event_shape>1, is it? We assume that this is what the user wanted and help her by reinterpreting the resulting batch dims as event dims. This is would not work with a N-D Gaussian because it could have both batch_shape and event_shape > 1. And therefore it is not in the set.

Reply:
But do you not *always* want to reinterpret *all* batch dimensions as event ones, *regardless* of pre-existing event dimensions? (i.e. you *never* want to have batch dimensions, or did I get it wrong?). If what I suggest doesn't make sense, then throw an error (i.e. when both batch and event dimension >1).

Reply:
I think doing that _always_ would be too much hidden service for the user. 

I thought batch_shape>1 and event_shape>1 is automatically covered by the `else` at the end? I could make it explicit though.

Reply:
We could also throw an error as soon as batch_shape>1 and **not** reinterpret using `Independent`. 
Then we would ask the user to use `CombinedJoint` to pass, e.g., a uniform prior over two parameters.

Reply:
I would prefer this last option, but with legal substances.